app/components/InteractContractModal/index.js

```diff
@@ -0,0 +1,722 @@
+    }
+  }
+
+  onUseMax = () => {
+    const { fee, gas, balance, storage } = this.state;
+    const max = balance - fee - gas - storage;
+    let amount = '0';
+    if (max > 0) {
+      amount = (max / utez).toFixed(6);
+    }
+    this.setState({ amount });
+  };
+
+  onUseMax1 = () => {
+    const { fee1, gas1, storage1 } = this.state;
+    const { addresses } = this.props;
+    const { balance } = addresses[0];
+    const max = balance - fee1 - gas1 - storage1;
+    let amount1 = '0';
+    if (max > 0) {
+      amount1 = (max / utez).toFixed(6);
+    }
+    this.setState({ amount1 });
+  };
+
+  onCloseClick = () => {
+    const { onCloseClick, addresses } = this.props;
+    this.setState({
+      ...defaultState,
+      selectedInvokeAddress: addresses[0].pkh,
+      balance: addresses[0].balance
+    });
+    onCloseClick();
+  };
+
+  onSetTab = activeTab => {
+    this.setState({ activeTab });
+  };
+
+  onChangeInvokeAddress = event => {
+    const { addresses } = this.props;
+    const pkh = event.target.value;
+    const address = addresses.find(address => address.pkh === pkh);
+    this.setState({ selectedInvokeAddress: pkh, balance: address.balance });
+  };
+
+  updatePassPhrase = passPhrase => this.setState({ passPhrase });
+
+  onLedgerConfirmation = val => {
+    const { activeTab } = this.state;
+    if (activeTab === 'invoke') {
+      this.setState({ isOpenLedgerConfirm: val });
+    } else {
+      this.setState({ isOpenLedgerConfirm1: val });
+    }
+  };
+
+  onEnterPress = (keyVal, isDisabled) => {
+    if (keyVal === 'Enter' && !isDisabled) {
+      this.onOperation();
+    }
+  };
+
+  openLink = element => openLinkToBlockExplorer(element);
+
+  onOperation = async () => {
+    const {
+      invokeAddress,
+      createNewAccount,
+      selectedParentHash,
+      setIsLoading,
+      isLedger,
+      addresses
+    } = this.props;
+
+    const {
+      activeTab,
+      smartAddress,
+      amount,
+      fee,
+      storage,
+      gas,
+      parameters,
+      selectedInvokeAddress,
+      passPhrase,
+      amount1,
+      gas1,
+      storage1,
+      fee1,
+      michelsonCode,
+      parameters1
+    } = this.state;
+
+    setIsLoading(true);
+
+    if (isLedger) {
+      this.onLedgerConfirmation(true);
+    }
+
+    let isOperationCompleted = false;
+    if (activeTab === 'invoke') {
+      const realParams = parameters ? JSON.parse(parameters) : null;
+      isOperationCompleted = await invokeAddress(
+        smartAddress,
+        fee,
+        amount,
+        storage,
+        gas,
+        realParams,
+        passPhrase,
+        selectedInvokeAddress,
+        selectedParentHash
+      ).catch(err => {
+        console.error(err);
+        return false;
+      });
+    } else {
+      const { pkh } = addresses[0];
+      const initStorage = parameters1 ? JSON.parse(parameters1) : {};
+      const initCode = michelsonCode ? JSON.parse(michelsonCode) : [];
```

Comment:
Deploying a contract without a code or storage is not possible. Deploy button should not be active if any of these fields are empty.